rdbms/src/test/java/io/spine/server/storage/jdbc/JdbcStorageFactoryTest.java

```diff
@@ -0,0 +1,255 @@
+        JdbcStorageFactory factory = JdbcStorageFactory.newBuilder()
+                                                       .setDataSource(dataSource)
+                                                       .setTypeMapping(MYSQL_5_7)
+                                                       .build();
+        ColumnTypeRegistry<?> registry = factory.getTypeRegistry();
+        assertNotNull(registry);
+    }
+
+    @Test
+    @DisplayName("generate single tenant view")
+    void generateSingleTenantView() {
+        DataSourceWrapper dataSource = GivenDataSource.withoutSuperpowers();
+        JdbcStorageFactory factory = JdbcStorageFactory.newBuilder()
+                                                       .setMultitenant(true)
+                                                       .setDataSource(dataSource)
+                                                       .setTypeMapping(MYSQL_5_7)
+                                                       .build();
+        assertTrue(factory.isMultitenant());
+        JdbcStorageFactory singleTenantFactory = factory.toSingleTenant();
+        assertFalse(singleTenantFactory.isMultitenant());
+    }
+
+    @Test
+    @DisplayName("use self as single tenant view")
+    void useSelfAsSingleTenantView() {
+        DataSourceWrapper dataSource = GivenDataSource.withoutSuperpowers();
+        JdbcStorageFactory factory = JdbcStorageFactory.newBuilder()
+                                                       .setMultitenant(false)
+                                                       .setDataSource(dataSource)
+                                                       .setTypeMapping(MYSQL_5_7)
+                                                       .build();
+        assertSame(factory, factory.toSingleTenant());
+    }
+
+    @Test
+    @DisplayName("use MySQL mapping by default")
+    void useMySqlMappingByDefault() {
+        DataSourceWrapper dataSource = GivenDataSource.whichIsStoredInMemory(newUuid());
+        JdbcStorageFactory factory = JdbcStorageFactory.newBuilder()
+                                                       .setMultitenant(false)
+                                                       .setDataSource(dataSource)
+                                                       .build();
+        assertEquals(MYSQL_5_7, factory.getTypeMapping());
+    }
+
+    private JdbcStorageFactory newFactory(boolean multitenant) {
+        return JdbcStorageFactory.newBuilder()
+                                 .setDataSource(config)
+                                 .setMultitenant(multitenant)
+                                 .setTypeMapping(MYSQL_5_7)
+                                 .build();
+    }
+
+    private static class TestEntity extends AbstractEntity<String, StringValue> {
+
+        private TestEntity(String id) {
+            super(id);
+        }
+    }
+
+    private static class TestAggregate extends Aggregate<String, StringValue, StringValueVBuilder> {
```

Comment:
Let's move the nested classes to `TestEnv`s, like you did for some of the cases already.